components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java

```diff
@@ -1030,17 +1034,24 @@ public ApplicationKeyMapping getApplicationKeyMapping(String consumerKey, String
                         api.setStatus(resultSet.getString("STATUS"));
                         api.setPolicy(resultSet.getString("API_TIER"));
                         api.setContext(resultSet.getString("CONTEXT"));
-                        String revision = resultSet.getString("REVISION_UUID");
-                        api.setIsDefaultVersion(isAPIDefaultVersion(connection, provider, name, version));
-                        api.setPolicy(getAPILevelTier(connection, apiUuid, revision));
-                        if (APIConstants.API_PRODUCT.equals(apiType)) {
-                            attachURlMappingDetailsOfApiProduct(connection, api);
-                        } else {
-                            if (deployment.equals(deploymentName)) {
-                                attachURLMappingDetails(connection, revision, api);
-                                apiList.add(api);
+                        String publishedDefaultApiVersion = resultSet.getString("PUBLISHED_DEFAULT_API_VERSION");
+                        if (StringUtils.isNotBlank(publishedDefaultApiVersion)) {
+                            api.setIsDefaultVersion(true);
+                        }
+                        if (isExpand) {
+                            String revision = resultSet.getString("REVISION_UUID");
+                            api.setPolicy(getAPILevelTier(connection, apiUuid, revision));
+                            if (APIConstants.API_PRODUCT.equals(apiType)) {
+                                attachURlMappingDetailsOfApiProduct(connection, api);
+                            } else {
+                                if (deployment.equals(deploymentName)) {
+                                    attachURLMappingDetails(connection, revision, api);
+                                }
                             }
+                        } else {
+                            api.setPolicy(null);
                         }
+                        apiList.add(api);
```

Comment:
Should this come under `if (deployment.equals(deploymentName)) {`. Need to return only APIs that matching the given deployment.

Reply:
done